### PR TITLE
Make pre/appended content be constrained by grid modifiers

### PR DESCRIPTION
**Pull request by Stanton**
This change makes prepended/appended content hidden on overflow so that grid widths affect the total with of the input including any prepended/appended content. 

## Before
Screenshot 2019-04-29 at 09 53 13

## After
Screenshot 2019-04-29 at 09 54 53

Change affects:
* form.date
* form.password
* form.range
* form.select
* form.select2
* form.text
* form.time

### CODE REVIEW

File: stylesheets/_component.forms.scss [hunk 1]
```diff
@@ -397,6 +397,7 @@ fieldset {
 .form__group.form__group .input-group .select2.select2,
 .form__group.form__group .input-group .form__control.form__control {
     flex: 1 1 auto;
+    overflow: hidden;
```
Comment (jamesjacobs) [inline stylesheets/_component.forms.scss]: Are you sure this is needed? In my quick tests the issue was fixed with `min-width: 0` and `overflow: hidden` on `.input-group-addon`

Comment (Stanton) [reply stylesheets/_component.forms.scss]: select2s completely broke at small widths without this, I'll provide screenshot

Comment (Stanton) [reply stylesheets/_component.forms.scss]: Before
Screenshot 2019-04-30 at 10 15 14

After
Screenshot 2019-04-30 at 10 13 20

Comment (jamesjacobs) [reply stylesheets/_component.forms.scss]: 👍

**Pull request merged**